packages/elements/src/calendar/index.ts

```diff
@@ -378,16 +455,44 @@ export class Calendar extends ControlElement implements MultiValue {
   */
   protected update (changedProperties: PropertyValues): void {
     if (!this.localMonthsNames || changedProperties.has(TranslatePropertyKey)) {
-      this.localMonthsNames = monthsNames(getLocale(this));
+      const locale = getLocale(this);
+      this.localMonthsNames = monthsNames(locale);
+      this.localFullMonthsNames = monthsNames(locale, 'long');
     }
     if (!this.localWeekdaysNames || changedProperties.has(TranslatePropertyKey)) {
-      this.localWeekdaysNames = weekdaysNames(getLocale(this));
+      const locale = getLocale(this);
+      this.localWeekdaysNames = weekdaysNames(locale);
+      this.localFullWeekdaysNames = weekdaysNames(locale, 'long');
     }
     this.shouldConstructFilters(changedProperties) && this.constructFilters();
 
     super.update(changedProperties);
   }
 
+  /**
+   * Called after render life-cycle finished
+   * @param changedProperties Properties which have changed
+   * @return {void}
+   */
+  protected updated (changedProperties: PropertyValues): void {
+    super.updated(changedProperties);
+
+    // This code is here to ensure that focus is not lost
+    // while navigating through the render views using keyboard
+    if (this.focused && changedProperties.has('renderView') && this.activeElement !== this.viewBtnRef.value) {
+      this.viewBtnRef.value!.focus();
```

Comment:
I know view button is always there, but asserting null throws a warning. Should we use `?` instead to only focus when button is really there ?

Reply:
Done